src/Bicep.Core/Semantics/Namespaces/INamespaceProvider.cs

```diff
@@ -3,13 +3,20 @@
 
 using System.Collections.Generic;
 using Bicep.Core.Features;
+using Bicep.Core.Syntax;
 using Bicep.Core.TypeSystem;
 
 namespace Bicep.Core.Semantics.Namespaces;
 
 public interface INamespaceProvider
 {
-    NamespaceType? TryGetNamespace(string providerName, string aliasName, ResourceScope resourceScope, IFeatureProvider features);
+    NamespaceType? TryGetNamespace(
+        string providerName,
+        string aliasName,
+        ResourceScope resourceScope,
+        IFeatureProvider features,
+        ImportDeclarationSyntax? ids = null
```

Comment:
>ids [](http://example.com/codeflow?start=33&length=3)

Since `providerName` and `aliasName` can be obtained from the `ids`, so we instead have an overload that accepts an `ImportDeclarationSyntax` instead of the first 2 params? (I wouldn't make that change if it would result in a bunch of if/else logic in the callers of these methods, though.)

Reply:
I think that we could do it as a separate refactor because it would result in a bunch of other code changes that are not really related to the change in behavior

Reply:
Yeah makes sense.